tests/core/basic20/task.yaml

```diff
@@ -58,4 +61,9 @@ execute: |
     done
 
     # ensure the "snap recovery" command works
-    snap recovery | MATCH '[0-9]+\ +canonical\*\ +ubuntu-core-20-[^ ]+\ +current'
+    MODEL="$(snap model --verbose | grep '^model' | awk '{ print $2 }')"
+    BRAND="$(snap model --verbose | grep '^brand' | cut -d' ' -f2- | xargs)"
+    if [ "$BRAND" = "canonical" ]; then
```

Comment:
actually looking at this, if you wanted to make this more generic to work for all possible brands you could essentially do something like:

```bash
brandid=$(snap model --verbose | grep brand-id: | awk '{print $2}')
if [ "$(snap known account "username=$brandid" | grep validation: | awk '{print $2}')" != "unproven" ]; then 
    brandid="$brandid\*"
fi
```

but what you have is fine for now

Reply:
Nice, I'll try that, thanks